common/test/java/com/couchbase/lite/CollectionTest.kt

```diff
@@ -0,0 +1,849 @@
+//
+// Copyright (c) 2022 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package com.couchbase.lite
+
+import com.couchbase.lite.internal.utils.Report
+import com.couchbase.lite.internal.utils.TestUtils
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNotSame
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import java.util.Locale
+
+class CollectionTest : BaseCollectionTest() {
+    //---------------------------------------------
+    //  Get Document
+    //---------------------------------------------
+    @Test
+    fun testGetNonExistingDocWithID() {
+        Assert.assertNull(testCollection!!.getDocument("non-exist"))
+    }
+
+    // get doc in the collection
+    @Test
+    @Throws(CouchbaseLiteException::class)
+    fun testGetExistingDocInCollection() {
+        val docId = "doc1"
+        createSingleDocInCollectionWithId(docId)
+        verifyGetDocumentInCollection(docId)
+    }
+
+    // get doc from the same collection from a different database instance
+    @Test
+    fun testGetExistingDocWithIDFromDifferentDBInstance() {
+        val docID = "doc1"
+        val doc = MutableDocument(docID)
+
+        val coll1 = baseTestDb.createCollection("coll_1")
```

Comment:
I do worry that many of these tests will fail, if one of them does.  We've had that happen a lot in the past.  I really do suggest using the name randomizer.